contentpacks/khanacademy.py

```diff
@@ -582,20 +583,67 @@ def retrieve_kalite_data(lang=en_lang_code, force=False, ka_domain=None, no_dubb
     ])
 
     url = lang_url.format(projection=json.dumps(projection), lang=lang, ka_domain=ka_domain)
-
     node_data_path = download_and_clean_kalite_data(url, lang=lang, ignorecache=force, filename="nodes.json")
 
     with open(node_data_path, 'r') as f:
         node_data = ujson.load(f)
 
+    # Create a list of projections to determine what is the current node data return from khan api.
+    youtube_ids = []
+    topic_title_list = []
+    exercise_ids = []
+    for node in node_data:
+        node_kind = node.get("kind")
+        if node_kind == NodeType.video:
+            if node["translated_youtube_lang"] == lang:
+                youtube_ids.append(node.get("youtube_id"))
+        if node_kind == NodeType.topic:
+            topic_title_list.append(node.get("title"))
+        if node_kind == NodeType.exercise:
+            exercise_ids.append(node.get("id"))
+
+    """
+    We must check if the languge code we specify has the same languge name
+        but defferent language code using get_lang_code_list function.
+    """
+    lang_list = get_lang_code_list(lang)
```

Comment:
I think this is where you need to check the value of the `multiple_lang_name` switch then proceed to loop thru the language codes list only if it is `True`, else, you use the value of the `lang` switch.